File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    
    vector<int> fibo_dp(n + 1, 0);
    fibo_dp[1] = 1;
    int& answer = fibo_dp[n];
    
    for(int i = 2 ; i <= n ; i++){
        fibo_dp[i] = (fibo_dp[i - 1] + fibo_dp[i - 2]) % 1234567;
    }
    
    return answer;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        
        int[] fibo_dp = new int[100000 + 1];
        fibo_dp[0] = 0; fibo_dp[1] = 1;
        
        for(int i = 2 ; i <= n ; i++) {
            fibo_dp[i] = (fibo_dp[i - 1] + fibo_dp[i - 2]) % 1234567;
        }
        
        
        return fibo_dp[n];
    }
}